app/components/modals/WithdrawModal.tsx

```diff
@@ -0,0 +1,267 @@
+'use client';
+
+import { useEffect, useState, useCallback } from 'react';
+
+interface WithdrawModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onComplete: () => void;
+  balance: number;
+  btcAddress: string;
+  lightningToken: string;
+}
+
+interface WithdrawQuote {
+  quantity: number;
+  fee: number;
+}
+
+type ModalState = 'loading' | 'quote' | 'processing' | 'success' | 'error';
+
+export default function WithdrawModal({
+  isOpen,
+  onClose,
+  onComplete,
+  balance,
+  btcAddress,
+  lightningToken
+}: WithdrawModalProps) {
+  const [modalState, setModalState] = useState<ModalState>('loading');
+  const [quote, setQuote] = useState<WithdrawQuote | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const fetchQuote = useCallback(async () => {
+    setModalState('loading');
+    setError(null);
+
+    try {
+      const response = await fetch('/api/lightning/withdraw/quote', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Lightning-Token': lightningToken
+        },
+        body: JSON.stringify({ l1_address: btcAddress })
+      });
+
+      if (!response.ok) {
+        const errorData = await response.json().catch(() => ({ error: response.statusText }));
+        throw new Error(errorData.error || 'Failed to get withdraw quote');
+      }
+
+      const data: WithdrawQuote = await response.json();
+      setQuote(data);
+      setModalState('quote');
+    } catch (err) {
+      console.error('Error fetching quote:', err);
+      setError(err instanceof Error ? err.message : 'Failed to get withdraw quote');
+      setModalState('error');
+    }
+  }, [btcAddress, lightningToken]);
+
+  // Fetch quote when modal opens
+  useEffect(() => {
+    if (isOpen) {
+      fetchQuote();
+    } else {
+      // Reset state when modal closes
+      setModalState('loading');
+      setQuote(null);
+      setError(null);
+    }
+  }, [isOpen, fetchQuote]);
+
+  // Close modal on escape key
+  useEffect(() => {
+    const handleEscKey = (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && modalState !== 'processing') {
+        onClose();
+      }
+    };
+
+    if (isOpen) {
+      window.addEventListener('keydown', handleEscKey);
+    }
+
+    return () => {
+      window.removeEventListener('keydown', handleEscKey);
+    };
+  }, [isOpen, onClose, modalState]);
+
```

Comment:
To fix `onClose` race condition:

```suggestion
useEffect(() => {
  return () => {
    if (successTimerRef.current) {
      clearTimeout(successTimerRef.current);
      successTimerRef.current = null;
    }
  };
}, []);

```